### PR TITLE
Port - PhotoHunter

### PR DESCRIPTION
**Pull request by StefanToncu**
Hi! I finished porting PhotoHunter from [gtk+](http://gcompris.net/screenshots-en.html#photohunter).
Please let me know what improvements i should add to the current version.

If someone wants to help with the animations, i will be very happy to accept.
[Phabricator task](https://phabricator.kde.org/T1535)

### CODE REVIEW

File: src/activities/photo_hunter/ActivityInfo.qml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+import GCompris 1.0
+
+ActivityInfo {
+  name: "photo_hunter/Photo_hunter.qml"
+  difficulty: 1
+  icon: "photo_hunter/photo_hunter.svg"
+  author: "Stefan Toncu &lt;stefan.toncu@cti.pub.ro&gt;"
+  demo: true
+  title: "Photo hunter"
```
Comment (petitlapin) [inline src/activities/photo_hunter/ActivityInfo.qml]: use qsTr() for string that are translatable

File: src/activities/photo_hunter/ActivityInfo.qml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  description: "Find the differences between the two pictures!"
+  //intro: "look at the two pictures carefully and click where you see a difference"
+  goal: "Visual Discrimination"
+  prerequisite: ""
+  manual: "Observe the two pictures carefully. There are some slight differences. When you find a difference you must click on it."
+  credit: ""
+  section: "fun"
+}
```
Comment (petitlapin) [inline src/activities/photo_hunter/ActivityInfo.qml]: there is a createdInVersion: 6000 missing

File: src/activities/photo_hunter/Photo_hunter.qml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            border.width: 2
+            border.color: "black"
+
+            color: "red"
+
+            GCText {
+                id: problemTxt
+                anchors.centerIn: parent
+                fontSize: items.cellSize / 7.5
+                text: qsTr("Click on the differences between the two images!   " + items.totalFound + "/" + items.total + " found.")
```
Comment (petitlapin) [inline src/activities/photo_hunter/Photo_hunter.qml]: use qsTr("text %1").arg(variable) when you have variables. Also, avoid having lines bigger than 80 characters if possible

File: src/activities/photo_hunter/Photo_hunter.qml [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            height: parent.height - background.barHeight - 30
+
+            anchors.top: problem.bottom
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.margins: 10
+
+            //left/top image
+            Observe {
+                id: img1
+                name: img2
```
Comment (petitlapin) [inline src/activities/photo_hunter/Photo_hunter.qml]: Why having img1 name img2 and the opposite below?

File: src/activities/photo_hunter/resource/README~ [hunk 5]
```diff
@@ -0,0 +1,10 @@
+https://openclipart.org/detail/17626/castle
```
Comment (petitlapin) [inline src/activities/photo_hunter/resource/README~]: this temporary file should not be commited

File: src/activities/photo_hunter/Photo_hunter.qml [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+            property int barHeightAddon: ApplicationSettings.isBarHidden ? 1 : 3
+            property int cellSize: Math.min(background.width / 11 , background.height / (9 + barHeightAddon))
+        }
+
+        onStart: { Activity.start(items) }
+        onStop: { Activity.stop() }
+
+        property bool vert: background.width < background.height
+        property double barHeight: (items.barHeightAddon == 1) ? 0 : 160
```
Comment (petitlapin) [inline src/activities/photo_hunter/Photo_hunter.qml]: using bar.height instead of 160 does not work?

**Comment by petitlapin** [comment]
Hi,

I have this error when finishing first level for first time: qrc:/gcompris/src/activities/photo_hunter/Photo_hunter.qml:99: TypeError: Cannot read property of null

Why do you have both levels in qml files and in photo_hunter.js? On my side, to add levels, it would be better to have them outside the main js file.

Johnny

**Comment by StefanToncu** [comment]
**All the previous comments have been addressed**.
I also made _**2 changes**_: 
- added **particles effect** when clicking on a difference
- added a field in levels: "radius", which will scale the blue circle (that marks the differences) to the specified "radius". It is **visible in level 3**, when clicking on the zebra

**Comment by petitlapin** [comment]
Hi,

when doing a bad click, maybe show a red cross?

Johnny

**Comment by StefanToncu** [comment]
yep, the red cross is here now.
I also added some "good sounds" for pressing on the differences and "bad sound" when pressing on anything else (when showing the red cross).

File: src/activities/photo_hunter/Photo_hunter.qml [hunk 7]
```diff
@@ -0,0 +1,172 @@
+/* GCompris - photo_hunter.qml
```
Comment (bdoin) [inline src/activities/photo_hunter/Photo_hunter.qml]: Naming conventions implies Qml file to by CamelCase so this one must be PhotoHunter.qml

File: src/activities/photo_hunter/resource/images/photo9.svg [hunk 8]
```diff
@@ -0,0 +1,1295 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (bdoin) [inline src/activities/photo_hunter/resource/images/photo9.svg]: You should put the images in resource/ not in a subdirectory

File: src/activities/photo_hunter/resource/README [hunk 9]
```diff
@@ -0,0 +1,11 @@
+https://openclipart.org/detail/17626/castle
```
Comment (bdoin) [inline src/activities/photo_hunter/resource/README]: We need to keep track of the reference on each resource file. For example we need to keep track of what the file photo1.svg refers to.

**Comment by bdoin** [comment]
As the images takes a lot of disk space, we could be smarter and have a single image per level. It could be done by creating a single svg with tags to mark items with differences (svg is an xml format). Then at runtime we load the first image normally and change it to have the tagged items to appear. The second image is then loaded from the first image plus xml modification to make the changes visible.

File: src/activities/photo_hunter/Observe.qml [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+                    id: particles
+                    clip: false
+                }
+            }
+
+            MouseArea {
+                id: mouseArea
+                anchors.fill: parent
+                onClicked: {
+                    img1.repeater.itemAt(index).particleLoader.active = active
```
Comment (bdoin) [inline src/activities/photo_hunter/Observe.qml]: It would be better to call a function on the items instead of being intrusive here.

**Pull request closed without merging**